Refactor image builds

CoryManson opened the pull request:
- Add Lakka 5.0
- Add Batocera 39
- Break buildroot builds out into their own shell scripts
- Decrease buildroot image sizes
- Switch to gzip from xz 
- Only upload images larger than 2GB to storage account to reduce cost
- Finally settle on a naming convention for images
- Remove unused packer builds
- Improve workflow
  - Add default release body text
  - Allow manual execution of workflow 
  - Write notice on workflow output when image larger than 2GB is found

CoryManson committed:
Add Lakka Image (#9)

CoryManson committed:
Adjust image name for retropie4.8-zero

CoryManson committed:
Improve workflow (#10)

othermod commented:
Your latest Zero 2 images for Kali and Batocera definitely aren't behaving. 
I'm getting immediate poweroff initially on the images, which I solved by moving to the separate files pi0.txt and cm4.txt. They definitely aren't getting parsed correctly. Maybe it's a size limit or something.
Here's my method:

This one applies to CM4 only. I do this because the Pi4 needs its own config, but if I do [pi4] then the cm4 also loads for config the pi4. I'll also have to specify the pi4 board-type when i make an image for it.
[board-type=0x14] 
include cm4.txt

this will load for the Pi0 and the Pi02
[pi0]
include pi0.txt

I pushed these separate config files to my repository and modified the config.txt for each os. I think there only needs to be a single pi0.txt and cm4.txt for each os to reference, since they all need the exact same info.

I still can't get a successful bootup on either image, but Kali they at least tries once the config.txt is modified. Haven't tried all of the other images yet.

CoryManson committed:
split config.txt into separate files (#11)

Co-authored-by: othermod <22604540+othermod@users.noreply.github.com>

CoryManson committed:
Merge branch 'main' into main

CoryManson commented:
awesome. Because I don't have a Zero 2 to test with, any zero builds are just a shot in the dark and require testing / changes so them not working is fully expected.

I'll add zero 2 images for all the OS to this PR today and we'll dial in the configs over the next little while before publishing. I'll try and get AP to help.

CoryManson committed:
Remove pi0 builds, add pi02 builds, update boot configs (#12)

* Remove zero images for rpios & retropie, add zero2 images for lakka & recalbox
* Upload cm4 & pi0
* Hopefully resolve regression with configs for cm4

CoryManson committed:
Add support for image checksums, fix Kali & RPiOS CM4 configs (#13)

TobiX committed:
Remove hardcoded firmware configurations for Batocera (#15)

This should be selected automatically AFAIK and the current setting
prevents a Zero2W from booting.

CoryManson committed:
Set FKMS V3D driver in config.txt (#16)

* Set FKMS V3D driver in config.txt
* update raspios source
+semver:minor

---------

Co-authored-by: cory manson <cory.manson@act.gov.au>

CoryManson committed:
Temporarily use prebuilt 32bit driver binaries for retropie (#18)

* Temporarily use prebuilt 32bit driver binaries for retropie
+semver:patch

---------

Co-authored-by: cory manson <cory.manson@act.gov.au>

CoryManson committed:
Skip creating release and uploading build artifact for non main branch builds (#21)

* Skips creating release and upload build artifact for non main builds

* Add artifact output for drivers & packer images

* revert upload artifact

+semver:patch

CoryManson committed:
Remove early azcopy step due to PR interference

TobiX committed:
Switch Batocera base image to bcm2837 (Pi3/Zero2) (#20)

Both, Batocera's bcm2836 & bcm2837 build are designed to support the
Zero2, but the differ in the shipped kernel: The bcm2836 image contains
a 32-bit kernel and the bcm2837 image contains a 64-bit kernel.

CoryManson committed:
Update lakka5 source image to nightly (#22)

* Update lakka5 source image to nightly

* update source image for lakka zero2

* Fix memory allocation for lakka zero2w

+semver:patch

CoryManson committed:
Allow drivers to be configured from boot partition (#23)

* Allow drivers to be configured from boot partition

* Don't set 64 bit for recalbox zero2w

* Change services to launch script that reads from config

* change to "enable_mouse"

* start osd after network

* add start script for kali

* remove detect arch

* add manual create release

* update conditions for creating release

* set boolean for createrelease

* update workflow call inputs

* allow services to persistn binaries

* Allow disabling the OSD service

* enable real time clock

+semver:minor

CoryManson committed:
Revert raspberry pi base image (#26)

+semver:patch

CoryManson committed:
Decomission Kali image (#27)

CoryManson commented:
This is ready for review. All images have been tested working on Pi Zero 2W & CM4 with the only exception being Retropie on the Zero 2W.

Kali was been removed which also allowed for removal of external storage requirements.
